packages/typespec-ts/test/modularIntegration/scalar.spec.ts

```diff
@@ -14,162 +14,108 @@ describe("Scalar Client", () => {
-    }
+    const result = await client.decimalType.requestBody(0.33333);
+    assert.isUndefined(result);
   });
 
   it("should get decimal request parameter", async () => {
-    try {
-      const result = await client.decimalType.requestParameter(0.33333);
-      assert.isUndefined(result);
-    } catch (err) {
-      assert.fail(err as string);
-    }
+    const result = await client.decimalType.requestParameter(0.33333);
+    assert.isUndefined(result);
   });
 
   it("should get decimal128 response body", async () => {
-    try {
-      const result = await client.decimal128Type.responseBody();
-      assert.strictEqual(result, 0.33333);
-    } catch (err) {
-      assert.fail(err as string);
-    }
+    const result = await client.decimal128Type.responseBody();
+    assert.strictEqual(result, 0.33333);
   });
 
   it("should put decimal128 request body", async () => {
-    try {
-      const result = await client.decimal128Type.requestBody(0.33333);
-      assert.isUndefined(result);
-    } catch (err) {
-      assert.fail(err as string);
-    }
+    const result = await client.decimal128Type.requestBody(0.33333);
+    assert.isUndefined(result);
   });
 
   it("should get decimal128 request parameter", async () => {
-    try {
-      const result = await client.decimal128Type.requestParameter(0.33333);
-      assert.isUndefined(result);
-    } catch (err) {
-      assert.fail(err as string);
-    }
+    const result = await client.decimal128Type.requestParameter(0.33333);
+    assert.isUndefined(result);
   });
 
   it("should fail to post decimal verify", async () => {
-    try {
-      const getResult = await client.decimalVerify.prepareVerify();
-      let total = 0;
-      getResult.forEach((decimal: number) => {
-        total += decimal;
-      });
-      await client.decimalVerify.verify(total);
-      assert.fail("Expected an exception to be thrown.");
-    } catch (err) {
-      assert.strictEqual(JSON.parse(JSON.stringify(err)).statusCode, 400);
-    }
+    const getResult = await client.decimalVerify.prepareVerify();
+    let total = 0;
+    getResult.forEach((decimal: number) => {
+      total += decimal;
+    });
+    total = Number(total.toFixed(1));
+    const result = await client.decimalVerify.verify(total);
+    assert.isUndefined(result);
   });
 
   it("should fail to post decimal128 verify", async () => {
-    try {
-      const getResult = await client.decimal128Verify.prepareVerify();
-      let total = 0;
-      getResult.forEach((decimal: number) => {
-        total += decimal;
-      });
-      await client.decimal128Verify.verify(total);
-      assert.fail("Expected an exception to be thrown.");
-    } catch (err) {
-      assert.strictEqual(JSON.parse(JSON.stringify(err)).statusCode, 400);
-    }
+    const getResult = await client.decimal128Verify.prepareVerify();
+    let total = 0;
+    getResult.forEach((decimal: number) => {
+      total += decimal;
+    });
+    total = Number(total.toFixed(1));
```

Comment:
please revert the changes for decimal and decimal128 ? they are suppose to fail for now.